pkg/cleanup/imagetags.go

```diff
@@ -2,40 +2,43 @@ package cleanup
 
 import (
 	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
 	"github.com/appuio/seiso/pkg/openshift"
 	imagev1 "github.com/openshift/api/image/v1"
 	log "github.com/sirupsen/logrus"
 	"github.com/thoas/go-funk"
-	"regexp"
-	"strings"
-	"time"
 )
 
 // MatchOption type defines how the tags should be matched
-type MatchOption int8
+type MatchOption string
 
 const (
-	MatchOptionDefault MatchOption = iota
-	MatchOptionExact
-	MatchOptionPrefix
+	// MatchOptionExact for exact matches
+	MatchOptionExact MatchOption = "exact"
+	// MatchOptionPrefix for prefix matches
+	MatchOptionPrefix MatchOption = "prefix"
 )
 
-// GetMatchingTags returns all tags matching one of the provided prefixes
-func GetMatchingTags(values, tags *[]string, matchOption MatchOption) []string {
+// GetMatchingTags returns all image tags matching one of the provided git tags
+func GetMatchingTags(gitTags, imageTags *[]string, matchOption MatchOption) []string {
 	var matchingTags []string
 
-	log.WithField("values", values).Debug("values")
-	log.WithField("tags", tags).Debug("tags")
-
-	if len(*values) > 0 && len(*tags) > 0 {
-		for _, value := range *values {
-			for _, tag := range *tags {
-				if match(tag, value, matchOption) {
-					matchingTags = append(matchingTags, tag)
-					log.WithFields(log.Fields{
-						"tag":   tag,
-						"value": value}).
-						Debug("Tag matched")
+	log.WithField("match", matchOption).
+		WithField("imageTags", imageTags).
+		WithField("gitTags", gitTags).
+		Debug("Matching imageTags with gitTags")
+
+	if len(*gitTags) > 0 && len(*imageTags) > 0 {
+		for _, gitTag := range *gitTags {
+			for _, imageTag := range *imageTags {
+				if match(imageTag, gitTag, matchOption) {
+					matchingTags = append(matchingTags, imageTag)
+					log.WithField("gitTag", gitTag).
+						WithField("imageTag", imageTag).
+						Debug("Found matching tag")
 				}
 			}
 		}
```

Comment:
This is very ugly to read... could this be a candidate for functional programming (e.g. funk library) or at least extracted to private methods with self-explaining names? or out of scope?

Reply:
I inverted the first if to return early. This removes one level of indentation at least...